build-logic/convention/build.gradle.kts

```diff
@@ -25,6 +27,12 @@ java {
     targetCompatibility = JavaVersion.VERSION_11
```

Comment:
you can raise the JavaVersion to 17. Here and in other places as well.

Reply:
Sounds good, raised to 17 everywhere.